### PR TITLE
Add service template link formatting

### PR DESCRIPTION
**Pull request by dangodev**
## Reason for change
In the DO rush we forgot to fully implement this. Custom service templates were technically emitting events, but the link couldn’t be formatted and it was left not completely finished.

This PR adds the ability to control where the custom service templates go, as well as adds tests to ensure the links are formatted correctly and the proper events are firing.

Screen Shot 2019-05-28 at 15 10 07

## Testing
1. In the Gatsby docs view, look at *Marketplace*
2. Copy + paste the following in your terminal: `document.addEventListener('manifold-template-click', e => console.log(e.detail.category))`
3. Click on a service card link. It should report the category in your console.

### CODE REVIEW

File: docs/docs/components/manifold-marketplace.md [hunk 1]
```diff
@@ -97,11 +97,12 @@ event (above). But it can also be turned into an `<a>` tag by specifying
 `link-format`:
 
 ```html
-<manifold-marketplace link-format="/product/:product" />
+<manifold-marketplace link-format="/product/:product" template-link-format="/template/:template" />
```
Comment (dangodev) [inline docs/docs/components/manifold-marketplace.md]: Open to feedback if there’s a better name for this 🤷‍♂

Comment (davidleger95) [reply docs/docs/components/manifold-marketplace.md]: I think this works.

Just a thought, take it or leave it: would it make sense to add `product-link-format` so the differentiation is more clear about what the link is for?

Comment (dangodev) [reply docs/docs/components/manifold-marketplace.md]: Hm. We use `link-format` for `<manifold-data-resource-list>` too, so I guess there are 2 options:

**Current way**
- `<manifold-marketplace link-format="">`
- `<manifold-service-card link-format="">`
- `<manifold-data-resource-list link-format="">`
- `<manifold-marketplace template-link-format="">`

**New way**
- `<manifold-marketplace product-link-format="">`
- `<manifold-service-card product-link-format="">`
- `<manifold-data-resource-list resource-link-format="">`
- `<manifold-marketplace template-link-format="">`

I could go either way. But I guess the new way does have advantages of being more future-proof

Comment (dangodev) [reply docs/docs/components/manifold-marketplace.md]: Updated to the new way. I think it’ll be clearer. LMK WYT ASAP PLS

Comment (davidleger95) [reply docs/docs/components/manifold-marketplace.md]: :shipit:

**Pull request merged**